hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/ratis/TestOzoneManagerDoubleBuffer.java

```diff
@@ -0,0 +1,201 @@
+        .setOzoneManagerRatisSnapShot(ozoneManagerRatisSnapshot)
+        .setmaxUnFlushedTransactionCount(1000)
+        .enableRatis(true)
+        .setIndexToTerm((i) -> 1L)
+        .build();
+
+    doNothing().when(omKeyCreateResponse).checkAndUpdateDB(any(), any());
+    doNothing().when(omBucketCreateResponse).checkAndUpdateDB(any(), any());
+    doNothing().when(omSnapshotCreateResponse1).checkAndUpdateDB(any(), any());
+    doNothing().when(omSnapshotCreateResponse2).checkAndUpdateDB(any(), any());
+
+    when(omKeyResponse.getTraceID()).thenReturn("keyTraceId");
+    when(omBucketResponse.getTraceID()).thenReturn("bucketTraceId");
+    when(omSnapshotResponse1.getTraceID()).thenReturn("snapshotTraceId-1");
+    when(omSnapshotResponse2.getTraceID()).thenReturn("snapshotTraceId-2");
+
+    when(omKeyResponse.getCreateSnapshotResponse()).thenReturn(null);
+    when(omBucketResponse.getCreateSnapshotResponse()).thenReturn(null);
+    when(omSnapshotResponse1.getCreateSnapshotResponse())
+        .thenReturn(snapshotResponse1);
+    when(omSnapshotResponse2.getCreateSnapshotResponse())
+        .thenReturn(snapshotResponse2);
+
+    when(omKeyCreateResponse.getOMResponse()).thenReturn(omKeyResponse);
+    when(omBucketCreateResponse.getOMResponse()).thenReturn(omBucketResponse);
+    when(omSnapshotCreateResponse1.getOMResponse())
+        .thenReturn(omSnapshotResponse1);
+    when(omSnapshotCreateResponse2.getOMResponse())
+        .thenReturn(omSnapshotResponse2);
+  }
+
+  @AfterEach
+  public void stop() {
+    if (doubleBuffer != null) {
+      doubleBuffer.stop();
+    }
+  }
+
+  private static Stream<Arguments> doubleBufferFlushCases() {
+    return Stream.of(
+        Arguments.of(Arrays.asList(omKeyCreateResponse,
+                omBucketCreateResponse),
+            1L, 2L, 1L, 2L, 2L, 2.0F),
+        Arguments.of(Arrays.asList(omSnapshotCreateResponse1,
+                omSnapshotCreateResponse1),
+            2L, 2L, 3L, 4L, 1L, 1.333F),
+        Arguments.of(Arrays.asList(omKeyCreateResponse,
+                omBucketCreateResponse,
+                omSnapshotCreateResponse1,
+                omSnapshotCreateResponse2),
+            3L, 4L, 6L, 8L, 2L, 1.333F),
+        Arguments.of(Arrays.asList(omKeyCreateResponse,
+                omSnapshotCreateResponse1,
+                omBucketCreateResponse,
+                omSnapshotCreateResponse2),
+            4L, 4L, 10L, 12L, 1L, 1.200F),
+        Arguments.of(Arrays.asList(omKeyCreateResponse,
+                omSnapshotCreateResponse1,
+                omSnapshotCreateResponse2,
+                omBucketCreateResponse),
+            4L, 4L, 14L, 16L, 1L, 1.142F)
+    );
+  }
+
+  @ParameterizedTest
+  @MethodSource("doubleBufferFlushCases")
+  public void testOzoneManagerDoubleBuffer(
+      List<OMClientResponse> omClientResponses,
+      long expectedFlushCounts,
+      long expectedFlushedTransactionCount,
+      long expectedFlushCountsInMetric,
```

Comment:
It's a bit surprising that the expectedFlushCounts don't match the expectedFlushCountsInMetric.  I'm guessing that the metrics are overall totals, while the counts get zeroed after each run.  Is that right?  If so, maybe add a comment to that effect.